src/redisearch_rs/slots_tracker/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+struct SlotsTrackerState {
+    /// Local responsibility slots - owned by this Redis instance in the cluster topology.
+    local: UnsafeCell<SlotSet>,
+    /// Fully available non-owned slots - locally available but not owned by this instance.
+    fully_available: UnsafeCell<SlotSet>,
+    /// Partially available non-owned slots - partially available and not owned by this instance.
+    partially_available: UnsafeCell<SlotSet>,
+    /// Version counter for tracking changes to the slots configuration.
+    /// Incremented whenever the slot configuration changes. Wraps around safely.
+    /// Atomic so it can be safely read from any thread.
+    version: AtomicU32,
+}
+
+// SAFETY: This is marked as Sync to allow use in static variables, but the caller
+// MUST ensure single-threaded access to the UnsafeCell fields. This is enforced
+// by the C API contract. The AtomicU32 is inherently Sync.
+unsafe impl Sync for SlotsTrackerState {}
+
+impl SlotsTrackerState {
+    const fn new() -> Self {
+        Self {
+            local: UnsafeCell::new(SlotSet::new()),
+            fully_available: UnsafeCell::new(SlotSet::new()),
+            partially_available: UnsafeCell::new(SlotSet::new()),
+            version: AtomicU32::new(0),
+        }
+    }
+}
+
+// ============================================================================
+// Global Static Instance (Private)
+// ============================================================================
+
+/// Global slots tracker state.
+///
+/// Contains all three slot sets and the version counter in a single structure
+/// for better organization and encapsulation.
+///
+/// # Safety
+///
+/// All mutation functions accessing this state are `unsafe extern "C"` and
+/// documented to require single-threaded access. Violating this contract
+/// (e.g., calling from multiple threads) is undefined behavior.
+static STATE: SlotsTrackerState = SlotsTrackerState::new();
+
+/// Reserved version value indicating unstable/partial availability.
+///
+/// This value will never equal a real version counter, so comparisons will always fail.
+/// Used when slots are available but partially or the coverage doesn't exactly match.
+pub const SLOTS_TRACKER_UNSTABLE_VERSION: u32 = u32::MAX;
+
+/// Reserved version value indicating slots are not available.
+///
+/// This value indicates that the query cannot proceed because required slots are not available.
+pub const SLOTS_TRACKER_UNAVAILABLE: u32 = u32::MAX - 1;
+
+/// Maximum valid version value before wrapping to 0.
+const MAX_VALID_VERSION: u32 = u32::MAX - 2;
+
+/// Increments the version counter, skipping reserved values.
+///
+/// If the current version is `MAX_VALID_VERSION` (u32::MAX - 2), wraps to 0
+/// instead of incrementing to reserved values (u32::MAX - 1 or u32::MAX).
+///
+/// # Safety
+///
+/// This function assumes single-threaded access (main thread only).
+fn increment_version() {
+    let current = STATE.version.load(Ordering::Relaxed);
```

Comment:
Can we attach the method to the struct? Like can we do implement increment_version for State? and then use these entry to call the static one? This would help in the testing right?